src/bulletin/bulletin.service.ts

```diff
@@ -69,4 +44,79 @@ export class BulletinService {
       }
     });
   }
+
+  async updateBulletin(
+    id: number,
+    updateBulletinDto: BulletinDTO,
+    pdfAttachments: Express.Multer.File[],
+  ) {
+    return this.prismaService.$transaction(async (tx) => {
+      try {
+        const { deleted_attachment_ids, ...bulletinData } = updateBulletinDto;
+        const updatedBulletin = await tx.bulletin.update({
+          where: { id },
+          data: {
+            ...bulletinData,
+          },
+        });
+        const attachments = [];
+        // Check if is there any deleted attachment ids to delete
+        // Reducing the query time by doing promise.all to simultaneously update the given ids
+        if (!isEmpty(deleted_attachment_ids)) {
+          const existingAttachments = await tx.pDFAttachment.findMany({
+            where: {
+              id: { in: deleted_attachment_ids },
+              deleted_at: null,
+              bulletin_id: id,
+            },
+          });
+          const existingAttachmentIdsSet = new Set(
+            existingAttachments.map((att) => att.id),
+          );
+          const nonExistingIds = deleted_attachment_ids.filter(
+            (id) => !existingAttachmentIdsSet.has(id),
+          );
+          if (!isEmpty(nonExistingIds)) {
+            throw new Error(
+              `Attachments with IDs ${nonExistingIds.join(
+                ', ',
+              )} do not exists to bulletin ID: ${id}`,
+            );
+          }
+
+          await Promise.all(
+            deleted_attachment_ids.map((deleteID) =>
+              tx.pDFAttachment.update({
```

Comment:
Is it possible to utilize updateMany() dito para di na need mag map through.

Reply:
alright I'll do that

Reply:
Woops, I tried doing that, but I realize that I need individual values on deleted_Attacment_ids so still need to iterate through all the items in the list.

I just did promise all to do it concurrently, to process it all at once.